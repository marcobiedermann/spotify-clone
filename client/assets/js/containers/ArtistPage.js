import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchArtist } from '../actions/artists';
import ArtistPage from '../pages/Artists/Artist';

const mapStateToProps = state => ({
  ...state,
  artist: state.artists.artist,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    fetchArtist,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ArtistPage);
