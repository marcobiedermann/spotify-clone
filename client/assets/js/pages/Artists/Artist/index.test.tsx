import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ArtistPage from '.';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Route render={(props) => <ArtistPage {...props} />} />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
