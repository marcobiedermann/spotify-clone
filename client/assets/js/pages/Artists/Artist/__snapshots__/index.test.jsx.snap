// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="error"
  >
    
  </div>
  <figure>
    <figcaption>
      <h2>
        
      </h2>
    </figcaption>
  </figure>
  <ul>
    <li>
      <a
        href="/albums"
        onClick={[Function]}
      >
        Albums
      </a>
    </li>
    <li>
      <a
        href="/related-artists"
        onClick={[Function]}
      >
        Related Artists
      </a>
    </li>
    <li>
      <a
        href="/top-tracks"
        onClick={[Function]}
      >
        Top Tracks
      </a>
    </li>
  </ul>
</div>
`;
